### PR TITLE
chore: Add set-up instructions

### PR DESCRIPTION
**Pull request by reginafcompton**
# Description

This PR adds explicit "getting started" instructions in the README.

### CODE REVIEW

**Comment by loganripplinger** [comment]
I will be tryinggg this now

**Comment by loganripplinger** [comment]
While following the instructions I got the following error,

```
.
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> 3ec1a5593d88, empty message
INFO  [alembic.runtime.migration] Running upgrade 3ec1a5593d88 -> e9bddc85352d, empty message
INFO  [alembic.runtime.migration] Running upgrade e9bddc85352d -> 4173e6fe893f, empty message
INFO  [alembic.runtime.migration] Running upgrade 4173e6fe893f -> 91fb9f00304b, empty message
INFO  [alembic.runtime.migration] Running upgrade 91fb9f00304b -> cee6c8486115, empty message
INFO  [alembic.runtime.migration] Running upgrade cee6c8486115 -> 6d1858d0da0d, empty message
INFO  [alembic.runtime.migration] Running upgrade 6d1858d0da0d -> 7998aa29b531, empty message
INFO  [alembic.runtime.migration] Running upgrade 7998aa29b531 -> 17af54db2ac2, empty message
INFO  [alembic.runtime.migration] Running upgrade 17af54db2ac2 -> 641292c41442, empty message
INFO  [alembic.runtime.migration] Running upgrade 641292c41442 -> 339d47a41f8c, empty message
INFO  [alembic.runtime.migration] Running upgrade 339d47a41f8c -> 4eea6d5ae1ee, empty message
INFO  [alembic.runtime.migration] Running upgrade 4eea6d5ae1ee -> 22122e1bd0e1, empty message
INFO  [alembic.runtime.migration] Running upgrade 22122e1bd0e1 -> 00081b33af3b, empty message
INFO  [alembic.runtime.migration] Running upgrade 00081b33af3b -> a82b1a6ec9ad, empty message
INFO  [alembic.runtime.migration] Running upgrade a82b1a6ec9ad -> 35f1e31ceb30, empty message
INFO  [alembic.runtime.migration] Running upgrade 35f1e31ceb30 -> 4d3d4feb8d41, empty message
INFO  [alembic.runtime.migration] Running upgrade 4d3d4feb8d41 -> 6eee1715ff21, empty message
INFO  [alembic.runtime.migration] Running upgrade 6eee1715ff21 -> 410130b93b0f, empty message
INFO  [alembic.runtime.migration] Running upgrade 410130b93b0f -> 49bfe630a344, empty message
INFO  [alembic.runtime.migration] Running upgrade 49bfe630a344 -> 1794e645d265, Seed AuthServer based on environment
Traceback (most recent call last):
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1685, in _execute_context
    self.dialect.do_executemany(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 903, in do_executemany
    context._psycopg2_fetched_rows = xtras.execute_values(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/psycopg2/extras.py", line 1292, in execute_values
    cur.execute(b''.join(parts))
psycopg2.errors.ForeignKeyViolation: insert or update on table "oauth2_clients" violates foreign key constraint "oauth2_clients_user_id_fkey"
DETAIL:  Key (user_id)=(ea6bbb45b05a4f7c9cbceb89589e083f) is not present in table "users".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/bin/flask", line 8, in <module>
    sys.exit(main())
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/flask/cli.py", line 967, in main
    cli.main(args=sys.argv[1:], prog_name="python -m flask" if as_module else None)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/flask/cli.py", line 586, in main
    return super(FlaskGroup, self).main(*args, **kwargs)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/flask/cli.py", line 426, in decorator
    return __ctx.invoke(f, *args, **kwargs)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/flask_migrate/cli.py", line 136, in upgrade
    _upgrade(directory, revision, sql, tag, x_arg)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/flask_migrate/__init__.py", line 96, in wrapped
    f(*args, **kwargs)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/flask_migrate/__init__.py", line 271, in upgrade
    command.upgrade(config, revision, sql=sql, tag=tag)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/command.py", line 294, in upgrade
    script.run_env()
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/script/base.py", line 490, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/util/pyfiles.py", line 97, in load_python_file
    module = load_module_py(module_id, path)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/util/compat.py", line 182, in load_module_py
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "migrations/env.py", line 96, in <module>
    run_migrations_online()
  File "migrations/env.py", line 90, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/runtime/environment.py", line 813, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/runtime/migration.py", line 560, in run_migrations
    step.migration_fn(**kw)
  File "/Users/loganripplinger/work/core/authserver/migrations/versions/1794e645d265_.py", line 216, in upgrade
    op.bulk_insert(oauth2_client_table, [
  File "<string>", line 8, in bulk_insert
  File "<string>", line 3, in bulk_insert
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/operations/ops.py", line 1980, in bulk_insert
    operations.invoke(op)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/operations/base.py", line 354, in invoke
    return fn(self, operation)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/operations/toimpl.py", line 173, in bulk_insert
    operations.impl.bulk_insert(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/ddl/impl.py", line 344, in bulk_insert
    self._exec(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/alembic/ddl/impl.py", line 146, in _exec
    return conn.execute(construct, multiparams)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1200, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 313, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1389, in _execute_clauseelement
    ret = self._execute_context(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1748, in _execute_context
    self._handle_dbapi_exception(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1929, in _handle_dbapi_exception
    util.raise_(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1685, in _execute_context
    self.dialect.do_executemany(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 903, in do_executemany
    context._psycopg2_fetched_rows = xtras.execute_values(
  File "/Users/loganripplinger/.local/share/virtualenvs/authserver-kU9Jyzj8/lib/python3.8/site-packages/psycopg2/extras.py", line 1292, in execute_values
    cur.execute(b''.join(parts))
sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) insert or update on table "oauth2_clients" violates foreign key constraint "oauth2_clients_user_id_fkey"
DETAIL:  Key (user_id)=(ea6bbb45b05a4f7c9cbceb89589e083f) is not present in table "users".

[SQL: INSERT INTO oauth2_clients (client_id, client_secret, client_id_issued_at, client_secret_expires_at, client_metadata, id, user_id) VALUES (%(client_id)s, %(client_secret)s, %(client_id_issued_at)s, %(client_secret_expires_at)s, %(client_metadata)s, %(id)s, %(user_id)s)]
[parameters: ({'client_id': 'd84UZXW7QcB5ufaVT15C9BtO', 'client_secret': 'cTQfd67c5uN9df8g56U8T5CwbF9S0LDgl4imUDguKkrGSuzI', 'client_id_issued_at': 1621011412, 'client_secret_expires_at': 0, 'client_metadata': '{"client_name": "M2M Client", "token_endpoint_auth_method": "client_secret_json", "grant_types": ["client_credentials"], "response_types": ["token"], "client_uri": "http://localhost:8000", "scope": ""}', 'id': '13c68e75d02a4c2280585f3a88549d39', 'user_id': 'ea6bbb45b05a4f7c9cbceb89589e083f'}, {'client_id': 'e84UZXW7QcB5ufaVT15C9BtO', 'client_secret': '', 'client_id_issued_at': 1621011412, 'client_secret_expires_at': 0, 'client_metadata': '{"client_name": "Public Client", "token_endpoint_auth_method": "none", "grant_types": ["authorization_code"], "response_types": ["code"], "client_uri": "http://localhost:8001", "redirect_uris": ["http://localhost:8001/auth/redirect"], "scope": ""}', 'id': '23c68e75d02a4c2280585f3a88549d3b', 'user_id': 'ea6bbb45b05a4f7c9cbceb89589e083f'})]
(Background on this error at: http://sqlalche.me/e/14/gkpj)
```

**Comment by reginafcompton** [comment]
@loganripplinger - I do not understand why we're seeing this now. 

The error originates in an old migration: https://github.com/brighthive/authserver/blob/b4f3fbeb38b6966211fd74735800e432c2cfeda1/migrations/versions/1794e645d265_.py

It might have something to do with the `env` or docker-compose adjustments.

**Comment by loganripplinger** [comment]
I set app_env to production and it seemed to work. https://github.com/brighthive/authserver/blob/b4f3fbeb38b6966211fd74735800e432c2cfeda1/migrations/versions/1794e645d265_.py#L68

File: docker-compose-devel.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
-version: '3'
+version: "3"
 services:
   postgres:
-    image: postgres:latest
+    image: postgres:12
```
Comment (johnosullivan) [inline docker-compose-devel.yml:4]: This worked for me

File: entrypoint.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ done
 
 if [ "$APP_ENV" == "DEVELOPMENT" ] || [ -z "$APP_ENV" ]; then
     export AUTHLIB_INSECURE_TRANSPORT=1
-    gunicorn -w 4 -b 0.0.0.0:8000 wsgi:app --reload --worker-class gevent --timeout 600
+    gunicorn -w 4 -b 0.0.0.0:10001 wsgi:app --reload --worker-class gevent --timeout 600
```
Comment (johnosullivan) [inline entrypoint.sh:17]: 👍

**Comment by loganripplinger** [comment]
So I had to set to APP_ENV PRODUCTION to get migrations to work. Then I set to DEVELOPMENT to match the health URL in the README.

**Review by loganripplinger — APPROVED** [review]
pls fix -- do we need conditionals in the migrations for app env?

**Review by loganripplinger — CHANGES_REQUESTED** [review]
http://0.0.0.0:10001/health

**Comment by reginafcompton** [comment]
> pls fix -- do we need conditionals in the migrations for app env?

I don't know. That seems like a separate ticket from this one. This PR simply adds some documentation, so you can get authserver up-and-running locally per the README.

**Review by loganripplinger — APPROVED** [review]

**Pull request merged**